lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -120,22 +159,26 @@ def _on_index_requested(self, event: IndexRequestedEvent) -> None:
         """
         if not self.unit.is_leader():
             return
-        if not self.opensearch.is_node_up():
+        if not self.opensearch.is_node_up() or not event.index:
             event.defer()
             return
 
+        # TODO extract to opensearch_index lib?
         try:
-            extra_user_roles = json.loads(event.extra_user_roles)
-        except (json.decoder.JSONDecodeError, TypeError) as err:
-            # TODO document what a client application would need to provide to make this work.
-            logger.error(err)
-            self.unit.status = BlockedStatus(ClientRelationBadRoleRequestMessage)
-            return
+            self.opensearch.request("PUT", event.index)
+        except OpenSearchHttpError as e:
+            if not (
+                e.response_code == 400
+                and e.response_body.get("error", {}).get("type")
+                == "resource_already_exists_exception"
+            ):
+                raise
 
         username = self._relation_username(event.relation)
         hashed_pwd, pwd = generate_hashed_password()
+        extra_user_roles = event.extra_user_roles if event.extra_user_roles else "default"
```

Comment:
Could we lowercase the extra_user_roles as a safeguard? 
```suggestion
        extra_user_roles = event.extra_user_roles.lower() if event.extra_user_roles else "default"
```